java/fory-core/src/main/java/org/apache/fory/type/Descriptor.java

```diff
@@ -432,10 +457,10 @@ public static SortedMap<Member, Descriptor> getAllDescriptorsMap(Class<?> clz) {
       });
 
   public static SortedMap<Member, Descriptor> getAllDescriptorsMap(
-      Class<?> clz, boolean searchParent) {
+      Class<?> clz, boolean searchParent, boolean globalRefTrackingEnabled) {
     try {
       Tuple2<SortedMap<Member, Descriptor>, SortedMap<Member, Descriptor>> tuple2 =
-          descCache.get(clz, () -> createAllDescriptorsMap(clz));
+          descCache.get(clz, () -> createAllDescriptorsMap(clz, globalRefTrackingEnabled));
```

Comment:
This is flaky, if `createAllDescriptorsMap` is invoked with `globalRefTrackingEnabled = true`, but later invoked with `globalRefTrackingEnabled=false`, then later invoke won't take efflect.

I prefer not passing parameters to `getAllDescriptorsMap/createAllDescriptorsMap`. We build Descriptor with default field annotation and whether it's primitive.

But when we get Descriptors from `TypeResolver/ClassResolver/XtypeResolver`, we create new `Descriptor` with correct settings. When we create new Descriptors, we should recompute based on `ForyField` annotation and `ref tracking` flag. 

Then in all serializers, we can take `trackingRef/nullable` as real truth.